docs/syntax.bnf

```diff
@@ -1,203 +1,128 @@
-  | "set" "prover" <ident>
-  | "set" "prover_timeout" <nat_lit>
-  | "assert" <assertion>
-  | "assertnot" <assertion>
-
-// Patterns for the rewrite tactic
-<rw_patt> ::=
-  | <term>
-  | "in" <term>
-  | "in" <ident> "in" <term>
-  | <ident> "in" <term>
-  | <term> "in" <ident> "in" <term>
-  | <term> "as" <ident> "in" <term>
-  
-// Proof tactics
-<tactic> ::=
-  | <query>
-  | "refine" <term>
-  | "assume" <ident>+
-  | "apply" <term>
-  | "simpl"
-  | "rewrite" "-"? {"[" <rw_patt> "]"}? <term>
-  | "reflexivity"
-  | "symmetry"
-  | "focus" <nat_lit>
-  | "why3" <string_lit>?
-  | "fail"
-
-// Builtins
-<builtin> ::=
-  | "0"
-  | "+1"
-  | "T"
-  | "P"
-  | "eq"
-  | "refl"
-  | "eq_ind"
-  | "top"
-  | "bot"
-  | "not"
-  | "or"
-  | "and"
-  | "imp"
-
-// Inductive types
-<inductive> ::= <ident> ":" <term> "≔" {"|" <ident> ":" <term>}*
-
-// Proofs
-<proof> ::= "begin" <tactic>* {"end"|"admit"|"abort"}
-
-// Unification rules
-<unif_rule> ::= <patt> "≡" <patt> "↪" <term> "≡" <term> {";" <term> "≡" <term>}*
-
-// Modifiers
-<property> ::= "constant" | "injective"
-<exposition> ::= "protected" | "private"
-<opacity> ::= "opaque"
-<modifier> ::= <opacity> | <exposition> | <property>
-
-// Commands
-<command> ::=
-  | <query>
-  | "require" <path> "as" <ident>
-  | "require" "open"? <path>
-  | "open" <path>
-  | <modifier>* "symbol" <ident> <arg>* {":" <type>}? {"≔" <term>}? proof?
-  | <exposition>? "inductive" <inductive> {"with" <inductive>}*
-  | "rule" <rule> {"with" <rule>}*
-  | "set" "declared" <string_lit>
-  | "set" "prefix" <float_lit> <string_lit> "≔" <qident>
-  | "set" "infix" {"left" | "right"}? <float_lit> <string_lit> "≔" <qident>
-  | "set" "quantifier" <qident>
-  | "set" "builtin" \"<builtin>\" "≔" <qident>
-  | "set" "unif_rule" <unif_rule>
-
-// Entry point: list of commands
-<main> ::= <command>*
+<ident> ::= ID
+
+<path> ::= ID (DOT ID)*
+
+<qident> ::= <path>
+
+<term_ident> ::= AT <qident>
+               | <qident>
+
+<patt> ::= DOLLAR <ident>
+         | DOLLAR WILD
+
+<arg_ident> ::= <ident>
+              | WILD
+
+<arg> ::= <arg_ident>
+        | L_PAREN <arg_ident>+ COLON <term> R_PAREN
+        | L_CU_BRACKET <arg_ident>+ [COLON <term>] R_CU_BRACKET
+
+<rw_patt_spec> ::= <term>
+                 | IN <term>
+                 | IN <ident> IN <term>
+                 | <term> IN <term> [IN <term>]
+                 | <term> AS <ident> IN <term>
+
+<rw_patt> ::= L_SQ_BRACKET <rw_patt_spec> R_SQ_BRACKET
+
+<tactic> ::= INTRO <arg_ident>+
+           | APPLY <term>
+           | SIMPL
+           | REWRITE [ASSOC] [<rw_patt>] <term>
+           | REFINE <term>
+           | REFL
+           | SYMMETRY
+           | WHY3 [STRINGLIT]
+           | <query>
+           | FAIL
+
+<ttactic> ::= <tactic> [SEMICOLON]
+
+<modifier> ::= CONSTANT
+             | INJECTIVE
+             | PROTECTED
+             | PRIVATE
+             | SEQUENTIAL
+             | OPAQUE
+
+<float_of_int> ::= FLOAT
+                 | INT
```

Comment:
Don't you mean float_or_int ("or" instead of "of")?

Reply:
In fact no, because the menhir rule takes a float or an int and returns a float, so it's either a float of an int or the identity.

Reply:
Ok but from a user point of view you can write a float or an int, not just an int as suggested by the name float_of_int!

Reply:
That's true, but I wonder whether this automated translation is what we want. The old BNF with tokens unfolded to their representation was better, even though we have to maintain it, I think.

Reply:
We could use a sed command on the generated file?